### PR TITLE
Test on same node versions as metalsmith/metalsmith

### LINKED ISSUE #62: Add .npmignore
For faster installs

### LINKED ISSUE #63: Support same node versions as core metalsmith lib
Currently we're not testing on the same node versions as [metalsmith](https://github.com/metalsmith/metalsmith/blob/master/.travis.yml). Should transpile our es6 with prepublish (http://mammal.io/articles/using-es6-today/).

### PR DESCRIPTION
**Pull request by ismay**
This requires transpiling our es6 before testing, and running the tests on the build results instead of the es6 code. It also means we should add a prepublish script for npm, and that we change the entry point for this lib to the build results.

Closes #63 
Closes #62

### CODE REVIEW

**Comment by ismay** [comment]
@doodzik What do you think? Unfortunately I don't have a solution for running eslint on travis as well with these versions of node. It doesn't work on node < 4, and enabling it conditionally is a bit of a hassle. I'd say we accept the regression in testing and open up a separate issue for running eslint on travis.

**Comment by doodzik** [comment]
I think it doesn't need to be an issue if eslint doesn't work on node 4, but still on the higher versions. Because we are testing the styling of the code and not the implementation on a platform.

File: test/index.js [hunk 1]
```diff
@@ -1,110 +1,118 @@
...
+ * Enable --harmony flag if needed
+ */
+
+require('harmonize')();
+
 /**
  * Dependencies
  */
 
-const equal = require('assert-dir-equal');
```
Comment (doodzik) [inline test/index.js]: Aren't we transpiling?
Why the change?

**Comment by ismay** [comment]
> I think it doesn't need to be an issue if eslint doesn't work on node 4, but still on the higher versions. Because we are testing the styling of the code and not the implementation on a platform.

Oh yeah definitely, but the problem is that I couldn't really find a good way of automatically running eslint on travis, but excluding incompatible node versions from running it. Otherwise it'll break the build on 0.12. If I could run eslint only on the latest node version that would be perfect.

File: test/index.js (hunk 1 above)
Comment (ismay) [inline test/index.js]: Ok, should be good now.

**Comment by doodzik** [comment]
can we run eslint only on one node version?
This would remove the headache a little.

**Comment by ismay** [comment]
Not through the `.travis.yml` I think (you set the node versions you want to test on, and it runs the tasks on all versions).

Maybe with something like https://www.npmjs.com/package/check-node-version. I couldn't get it to work in this PR, but with some fiddling around it should be possible.
